internal/gitlab/gitlab.go

```diff
@@ -538,6 +545,10 @@ func MRUnapprove(pid interface{}, id int) error {
 	if resp != nil && resp.StatusCode == http.StatusForbidden {
 		return ErrStatusForbidden
 	}
+	if resp != nil && resp.StatusCode == http.StatusNotFound {
+		// returns 404 if the MR has already been approved
```

Comment:
s/approved/unapproved maybe? :)

Just curious, the API returns 401 when it was already approved and 404 when already unapproved? two different error codes for basically the same operation? Don't get me wrong, I'm not doubting or anything like that, I'm just surprised how odd it is.... even more using 404 (not found??). lol.

Reply:
Unapprove makes sense. I'll change that comment.

The 401/404 makes sense.  Think of this it this way: in order to have an unapprove available the MR must be in the approve state.  If the MR is in the unapprove state then the unapprove does not exist -- hence 404 (Not Found).  If an MR is approved, then someone attempts to approve it again, then they should be blocked from approving a second time -- hence 401 (Unauthorized).

Reply:
Do you think I should return "fail" (ie "1") in these cases instead of success?  I mean -- the command to approve/unapprove didn't actually do anything and that should be a failure, no?